mods/ctf/ctf_modebase/markers.lua

```diff
@@ -158,46 +160,61 @@ local function marker_func(name, param, specific_player)
 	if not pointed then
 		return false, "Can't find anything to mark, too far away!"
 	end
-
-	local message = string.format("m [%s]: %s", name, param)
-	local pos
-
-	if pointed.type == "object" then
-		local concat
-		local obj = pointed.ref
-		local entity = obj:get_luaentity()
-
-		-- If object is a player, append player name to display text
-		-- Else if obj is item entity, append item description and count to str.
-		if obj:is_player() then
-			concat = obj:get_player_name()
-		elseif entity then
-			if entity.name == "__builtin:item" then
-				local stack = ItemStack(entity.itemstring)
-				local itemdef = minetest.registered_items[stack:get_name()]
-
-				-- Fallback to itemstring if description doesn't exist
-				-- Only use first line of itemstring
-				concat = string.match(itemdef.description or entity.itemstring, "^([^\n]+)")
-				concat = concat .. " " .. stack:get_count()
+	message = string.format("m [%s]: %s", name, param)
+
+		if pointed.type == "object" then
+			local concat
+			local obj = pointed.ref
+			local entity = obj:get_luaentity()
+
+			-- If object is a player, append player name to display text
+			-- Else if obj is item entity, append item description and count to str.
+			if obj:is_player() then
+				concat = obj:get_player_name()
+			elseif entity then
+				if entity.name == "__builtin:item" then
+					local stack = ItemStack(entity.itemstring)
+					local itemdef = minetest.registered_items[stack:get_name()]
+					-- Fallback to itemstring if description doesn't exist
+					-- Only use first line of itemstring
+					concat = string.match(itemdef.description or entity.itemstring, "^([^\n]+)")
+					concat = concat .. " " .. stack:get_count()
+				end
+			end
+				pos = obj:get_pos()
+			if concat then
+				message = message .. " <" .. concat .. ">"
 			end
+		else
+			pos = pointed.under
 		end
 
-		pos = obj:get_pos()
-		if concat then
-			message = message .. " <" .. concat .. ">"
-		end
-	else
-		pos = pointed.under
+	if vector.distance(pos, player:get_pos()) then
+		hpmarker = true
+	end
+	if hpmarker then
+		message = string.format("m [%s]: HP=%i/%i", name, player:get_hp(), player:get_properties().hp_max)
+		pos = player:get_pos()
 	end
 
 	ctf_modebase.markers.add(name, message, pos, nil, specific_player)
 
 	marker_cooldown:set(name, MARKER_PLACE_INTERVAL)
 
-	return true, "Marker is placed!"
+	if hpmarker then
+		return true, "HP marker is placed!"
+	else
+		return true, "Marker is placed!"
+	end
 end
 
+minetest.register_chatcommand("mhp", {
+	description = "Place a HP marker in your look direction",
+	params = "",
+	privs = {interact = true, shout = true},
+	func = function(name, param, specific_player) return marker_func(name, param, specific_player, true) end
```

Comment:
The `func` function of chatcommands only passes two arguments, did you test this code?

Reply:
IIRC that's how Farooq had made the marker changes to be to make the function reusable. There's a part of me still I'm confusion about this 😅

Reply:
```suggestion
	func = function(name, param) return marker_func(name, param, nil, true) end
```

@src4026 did it wrong but because in Lua an argument not provided is `nil`, it didn't cause any problem and it works. This suggestion makes it better.

Reply:
Sorry for that, commiting suggestion.